dspace-server-webapp/src/test/java/org/dspace/app/rest/CommunityCollectionParentIT.java

```diff
@@ -0,0 +1,218 @@
+    public void col2ParentCommunityTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token)
+            .perform(get("/api/core/collections/" + col2.getID() + "/parentCommunity"))
+            .andExpect(jsonPath("$", is(CommunityMatcher.matchCommunityEntry(communityAA.getName(), communityAA.getID(),
+                                                                             communityAA.getHandle()))))
+            .andExpect(jsonPath("$", not(is(CommunityMatcher
+                                                .matchCommunityEntry(communityA.getName(), communityA.getID(),
+                                                                     communityA.getHandle())))))
+            .andExpect(jsonPath("$", not(is(CommunityMatcher
+                                                .matchCommunityEntry(communityAB.getName(), communityAB.getID(),
+                                                                     communityAB.getHandle())))));
+
+    }
+
+    @Test
+    public void col3ParentCommunityTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token)
+            .perform(get("/api/core/collections/" + col3.getID() + "/parentCommunity"))
+            .andExpect(jsonPath("$", is(CommunityMatcher.matchCommunityEntry(communityAB.getName(), communityAB.getID(),
+                                                                             communityAB.getHandle()))))
+            .andExpect(jsonPath("$", not(is(CommunityMatcher
+                                                .matchCommunityEntry(communityA.getName(), communityA.getID(),
+                                                                     communityA.getHandle())))))
+            .andExpect(jsonPath("$", not(is(CommunityMatcher
+                                                .matchCommunityEntry(communityAA.getName(), communityAA.getID(),
+                                                                     communityAA.getHandle())))));
+
+    }
+
+    @Test
+    public void comAAParentCommunityTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token)
+            .perform(get("/api/core/communities/" + communityAA.getID() + "/parentCommunity"))
+            .andExpect(jsonPath("$", Matchers
+                .is(CommunityMatcher.matchCommunityEntry(communityA.getID(), communityA.getHandle()))))
+            .andExpect(jsonPath("$", Matchers
+                .not(Matchers.is(CommunityMatcher.matchCommunityEntry(communityB.getID(), communityB.getHandle())))));
+
+    }
+
+    @Test
+    public void comAParentCommunityTest() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+        getClient(token).perform(get("/api/core/communities/" + communityA.getID() + "/parentCommunity"))
+                        .andExpect(status().isNoContent());
+
+
+    }
+
```

Comment:
we need all the test related to the security and wrong scenarios. 
The specified item, collection, community 
- don't exist, don't make sense (i.e. it is not an uuid) -> (404)
- are not visibile (401, 403)
- the starting object is visible but the parent is not visible (401, 403)

Reply:
@abollini : I'm not sure the 404 test is needed, as we already have corresponding 404 tests in ITs for Communities, Collections & Items. How can a 404 test here add anything different?

I do agree that the other two types of tests would be worth adding, though we could consider doing so as a separate security ticket (as you noted), provided that we create a ticket in JIRA.

Reply:
obvious test are useful for the future because they look obvious now that we know the implementation but if we go trough a refactoring the implementation could be completely different and break something that we expect to just work. So I always recommend to make black box test, indeed we should follow a TDD approach that would imply that.

Anyway, I'm not completely sure that it will return 404 right now, maybe we can get a NPE o ClassCastException so a 500 see https://github.com/DSpace/DSpace/pull/2673/files#diff-59c10eaa99f8fbc2322484a9bacff5afR51 (again I'm just guessing but it is wrong to look to the communityservice to get the answer, we need to have test to proof it)

Reply:
it has been addressed